src/components/data/DataNavigation.js

```diff
@@ -0,0 +1,460 @@
+ * @param {string} path -  CyVerse Data Store path
+ * @param {string} userHomePath - Users home folder path
+ * @param {string} userTrashPath - Users trash folder path
+ * @returns {string} relativePath - relative path to CyVerse Data Store e.g: /analyses/wordcount/logs
+ */
+function getRelativePath(path, userHomePath, userTrashPath) {
+    let relativePath = "";
+    if (path.indexOf(userHomePath) !== -1) {
+        relativePath = path.replace(userHomePath, "");
+    } else if (path.indexOf(userTrashPath) !== -1) {
+        relativePath = path.replace(userTrashPath, "");
+    } else if (path.indexOf(constants.COMMUNITY_DATA_PATH) !== -1) {
+        relativePath = path.replace(constants.COMMUNITY_DATA_PATH, "");
+    } else if (path.indexOf(constants.SHARED_WITH_ME_PATH) !== -1) {
+        relativePath = path.replace(constants.SHARED_WITH_ME_PATH, "");
+    }
+    return relativePath;
+}
+
+function FolderSelectorMenu({
+    root,
+    path,
+    userHomePath,
+    userTrashPath,
+    baseId,
+    intl,
+}) {
+    const classes = useStyles();
+    const router = useRouter();
+    const [anchorEl, setAnchorEl] = useState(null);
+    const relativePath = getRelativePath(path, userHomePath, userTrashPath);
+    const pathItems = getPathItems(relativePath);
+    const [selectedIndex, setSelectedIndex] = useState(0);
+
+    useEffect(() => {
+        console.log(
+            "selected index=>" + pathItems.length > 0 ? pathItems.length - 1 : 0
+        );
+        setSelectedIndex(pathItems.length > 0 ? pathItems.length - 1 : 0);
+    }, [pathItems]);
+
+    const handleClickListItem = (event) => {
+        setAnchorEl(event.currentTarget);
+    };
+
+    const handleMenuItemClick = (event, index) => {
+        setSelectedIndex(index);
+        setAnchorEl(null);
+        const relativePath = getRelativePath(path, userHomePath, userTrashPath);
+        router.push(pathToRoute(root, relativePath, index));
+    };
+
+    const handleClose = () => {
+        setAnchorEl(null);
+    };
+
+    return (
+        pathItems.length > 0 && (
+            <>
+                <List
+                    component="nav"
+                    aria-controls={formatMessage(
+                        intl,
+                        "selectedFolderAriaControl"
+                    )}
+                    aria-label={formatMessage(intl, "selectedFolderAriaLabel")}
+                    id={build(baseId, ids.PATH_ITEMS)}
+                >
+                    <ListItem
+                        button
+                        aria-haspopup="true"
+                        aria-controls={formatMessage(
+                            intl,
+                            "selectedFolderAriaMenuItemControl"
+                        )}
+                        aria-label={pathItems[selectedIndex]}
+                        onClick={handleClickListItem}
+                        className={classes.selectedListItem}
+                    >
+                        <>
```

Comment:
I think `ListItem` can already have multiple children, so this fragment wrapper may not be needed.